packages/payment-processor/src/payment/swap-conversion-erc20.ts

```diff
@@ -0,0 +1,79 @@
+import { ContractTransaction, providers, Signer, BigNumberish } from 'ethers';
+
+import { erc20SwapConversionArtifact } from '@requestnetwork/smart-contracts';
+import { ClientTypes, PaymentTypes } from '@requestnetwork/types';
+
+import { ITransactionOverrides } from './transaction-overrides';
+import { getProvider, getSigner } from './utils';
+import { checkErc20Allowance, encodeApproveAnyErc20 } from './erc20';
+
+/**
+ * Processes the approval transaction of a given payment ERC20 to be spent by the swap router,
+ * if the current approval is missing or not sufficient.
+ * @param request request to pay, used to know the network
+ * @param ownerAddress address of the payer
+ * @param paymentCurrency ERC20 currency used for the swap
+ * @param signerOrProvider the web3 provider. Defaults to Etherscan.
+ * @param minAmount ensures the approved amount is sufficient to pay this amount
+ * @param overrides optionally, override default transaction values, like gas.
+ */
+export async function approveErc20ForSwapWithConversionIfNeeded(
+  request: ClientTypes.IRequestData,
+  ownerAddress: string,
+  paymentTokenAddress: string,
+  signerOrProvider: providers.Provider | Signer = getProvider(),
+  minAmount: BigNumberish,
+  overrides?: ITransactionOverrides,
+): Promise<ContractTransaction | void> {
+  const network =
+    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network ||
+    'mainnet';
```

Comment:
```suggestion
  if (!request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY]) {
    throw new Error(`The request must have the payment network any-to-erc20-proxy`);
  }
  const network =
    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network ||
    'mainnet';
```

Reply:
I'm not a big fan of default network values, especially within the code. It can play tricks while testing and potentially hide differences between `mainnet` and `rinkeby`.
```suggestion
    if (!request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY]) {
	    throw new Error(`The request must have the payment network any-to-erc20-proxy`);
	  }
	  const network =	    request.extensions[PaymentTypes.PAYMENT_NETWORK_ID.ANY_TO_ERC20_PROXY].values.network;
```